### PR TITLE
railway-wallet: init at 5.17.0

### PR DESCRIPTION
**Pull request by mitchmindtree**
## Description of changes

This is a quick wrapper around the Railway Wallet AppImage - a private DeFi wallet for Linux based on the Railgun protocol.

- https://www.railway.xyz/
- https://www.railgun.org/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ra/railway-wallet-bin/package.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+appimageTools.wrapType2 rec {
+  pname = "railway-wallet-bin";
+  version = "5.16.9";
+
+  src = fetchurl {
+    url = "https://github.com/Railway-Wallet/Railway-Wallet/releases/download/v${version}/Railway-linux-x86_64.AppImage";
+    hash = "sha256-UI/eJoQFtWzP/vMumJVp8BRylmUqRMTdbAJpFYDzsh8=";
+  };
+
+  meta = with lib; {
```
Comment (heisfer) [inline pkgs/by-name/ra/railway-wallet-bin/package.nix:15]: `with lib;` is problematic antipattern. Use `lib.licenses` `with lib.maintainers;` below.

Comment (mitchmindtree) [reply pkgs/by-name/ra/railway-wallet-bin/package.nix]: Resolved in https://github.com/NixOS/nixpkgs/pull/320395/commits/0eccf2dee2ec9270e6be5a1551970fc8577ad585

**Comment by mitchmindtree** [comment]
cc @heisfer this should be ready for re-review when you get a chance :pray:

**Comment by mitchmindtree** [comment]
Just updated from `5.16.9` to `5.17.0`, amended the commit and updated the PR title.

**Comment by mitchmindtree** [comment]
Looks like the new formatting lint was just added! I've amended the commit to follow the new standard.

**Comment by mitchmindtree** [comment]
I just noticed they re-licensed from GPL-3-only to AGPL-3-only in their latest version. I've updated the license field and amended the commit.

File: pkgs/by-name/ra/railway-wallet-bin/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/ra/railway-wallet-bin/package.nix:1]: Since there is no none bin variant of this packages, we don't add a -bin suffix in nixpkgs

Comment (mitchmindtree) [reply pkgs/by-name/ra/railway-wallet-bin/package.nix:1]: Thanks @SuperSandro2000, I've removed the `-bin` suffix from the package name, package directory and amended the commit message.

**Pull request merged**